scripts/vote_2024_xx_xx_upgrade_optimism_bridge.py

```diff
@@ -0,0 +1,156 @@
+L2_TOKEN_RATE_ORACLE_NEW_IMPL: str = "0x6b314986E3737Ce23c2a13036e77b3f5A846F8AF"
+
+def encode_upgrade_call(proxy_admin: str, proxy: str, new_impl: str):
+    scroll_executor = interface.OpBridgeExecutor(
+        "0x6625c6332c9f91f2d27c304e729b86db87a3f504"
+    )  # any address to bypass
+
+    params = eth_abi.encode(["address", "address"], [proxy, new_impl])
+
+    return scroll_executor.queue.encode_input([proxy_admin], [0], ["upgrade(address,address)"], [params], [False])
+
+def encode_l1_l2_sendMessage(to: str, fee: int, calldata: str):
+    l1_l2_msg_service = interface.OpCrossDomainMessenger(OPTIMISM_L1_L2_MESSAGE_SERVICE)
+
+    return l1_l2_msg_service.sendMessage.encode_input(to, fee, calldata, 1_000_000)
+
+def start_vote(tx_params: Dict[str, str], silent: bool) -> bool | list[int | TransactionReceipt | None]:
+    """Prepare and run voting."""
+
+    if not network_name() in ("sepolia", "sepolia-fork"):
+        return
+
+    l1_token_bridge_proxy = interface.OssifiableProxy(L1_TOKEN_BRIDGE_PROXY);
+
+    call_script_items = [
+            # 1. L1 Bridge
+            agent_forward(
+            [
+                (
+                    l1_token_bridge_proxy.address,
+                    l1_token_bridge_proxy.proxy__upgradeTo.encode_input(L1_TOKEN_BRIDGE_NEW_IMPL),
+                )
+            ]),
+            # 2. L2 Bridge
+            agent_execute(
+                OPTIMISM_L1_L2_MESSAGE_SERVICE,
+                10**17,
+                encode_l1_l2_sendMessage(
+                    OPTIMISM_BRIDGE_EXECUTOR,
+                    0,
+                    encode_upgrade_call(PROXY_ADMIN_ADDR, L2_TOKEN_BRIDGE_PROXY, L2_TOKEN_BRIDGE_NEW_IMPL),
+                )
+            ),
+            # 3. wstETH on L2
+            agent_execute(
+                OPTIMISM_L1_L2_MESSAGE_SERVICE,
+                10**17,
+                encode_l1_l2_sendMessage(
+                    OPTIMISM_BRIDGE_EXECUTOR,
+                    0,
+                    encode_upgrade_call(PROXY_ADMIN_ADDR, L2_NON_REBASABLE_TOKEN_PROXY, L2_NON_REBASABLE_TOKEN_NEW_IMPL),
+                )
+            ),
+            # 4. stETH on L2
+            agent_execute(
+                OPTIMISM_L1_L2_MESSAGE_SERVICE,
+                10**17,
+                encode_l1_l2_sendMessage(
+                    OPTIMISM_BRIDGE_EXECUTOR,
+                    0,
+                    encode_upgrade_call(PROXY_ADMIN_ADDR, L2_REBASABLE_TOKEN_PROXY, L2_REBASABLE_TOKEN_NEW_IMPL),
+                )
+            ),
+            # 5. TokenRateOracle
+            agent_execute(
+                OPTIMISM_L1_L2_MESSAGE_SERVICE,
+                10**17,
+                encode_l1_l2_sendMessage(
+                    OPTIMISM_BRIDGE_EXECUTOR,
+                    0,
+                    encode_upgrade_call(PROXY_ADMIN_ADDR, L2_TOKEN_RATE_ORACLE_PROXY, L2_TOKEN_RATE_ORACLE_NEW_IMPL),
+                )
+            )
+    ]
+
+    vote_desc_items = [
```

Comment:
I guess there should be an item to upgrade the `LidoLocator` impl (include a newly deployed `postTokenRebaseReceiver`)